src/treeview/writer.js

```diff
@@ -0,0 +1,623 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Position from './position.js';
+import Element from './element.js';
+import Text from './text.js';
+import Range from './range.js';
+import CKEditorError from '../ckeditorerror.js';
+
+/**
+ * Tree model Writer class.
+ * Writer defines a high-level API for TreeView manipulations.
+ *
+ * @memberOf core.treeView
+ */
+ export default class Writer {
+	constructor() {
+		/**
+		 * Priorities map. Maps node to priority.
+		 * Nodes with priorities are considered as attributes.
+		 *
+		 * @member {WeakMap} core.treeView.Writer#_priorities
+         * @protected
+         */
+		this._priorities = new WeakMap();
+	}
+
+	/**
+	 * Returns `true` if provided node is a `container` node.
+	 * Node is considered as `container` when it is a {@link core.treeView.Element Element} instance and has no priority
```

Comment:
It would be cool to explain that the container is. The fact it has no priority is not a point. The point is method will break attribute, but not break containers, nor enter them.